grakn-graph/src/main/java/ai/grakn/graph/internal/RoleTypeImpl.java

```diff
@@ -124,6 +154,10 @@ public void innerDelete(){
             throw new ConceptException(ErrorMessage.CANNOT_DELETE.getMessage(getName()));
         } else {
             super.innerDelete();
+
+            //Clear all internal caching
+            cachedRelationTypes.clear();
+            cachedDirectPlayedByTypes.clear();
```

Comment:
You also need to alert all connected relation types and played-by types that the role has been deleted (so they can clear their cache).